src/icons/scripts/assertNoDuplicates.ts

```diff
@@ -0,0 +1,98 @@
+import fs from "fs";
+import path from "path";
+import { formatComponentName } from "./formatComponentName";
+
+const SVG_PATH = path.resolve(__dirname, "..", "svgs");
+
+const streamlineDuplicates: Array<[string, string]> = [];
+const otherCategoryDuplicates: Array<[string, string]> = [];
+
+const categorizedFilenames = fs
+  .readdirSync(SVG_PATH)
+  .filter((dirname) => !dirname.startsWith("."))
+  .reduce<Array<{ category: string; filename: string; componentName: string }>>(
+    (accumulator, categoryDirectory) => {
+      const iconFilenames = fs
+        .readdirSync(path.join(SVG_PATH, categoryDirectory))
+        .filter((dirname) => !dirname.startsWith("."));
+      return accumulator.concat(
+        iconFilenames.map((filename) => ({
+          category: categoryDirectory,
+          filename: path.join(categoryDirectory, filename),
+          componentName: formatComponentName(
+            path.basename(filename, path.extname(filename))
+          ),
+        }))
+      );
+    },
+    []
+  );
+
+categorizedFilenames.forEach(
+  ({ category, filename, componentName }, index, list) => {
+    const duplicate = list
+      .slice(index + 1)
+      .find((item) => item.componentName === componentName);
+
+    if (duplicate) {
+      if (duplicate.category === category) {
+        // Same category, must be streamline
+        streamlineDuplicates.push([filename, duplicate.filename]);
+      } else {
+        otherCategoryDuplicates.push([filename, duplicate.filename]);
```

Comment:
This could still be a streamline duplicate right? I feel like the `streamlineDuplicates` message would still be more helpful in that case?

Reply:
Yeah you're right, it could still be a streamline duplicate.